File: Companies/VMware/SquaresofSortedArray.cpp
#include <iostream>
#include <vector>

using namespace std;

class SquaresofSortedArray
{
public:
  vector<int> sortedSquares(vector<int> &A)
  {
    vector<int> re(A.size());
    int l = 0, r = A.size() - 1;
    for (int i{r}; i >= 0; i--)
    {
      re[i] = abs(A[l]) >= abs(A[r]) ? A[l] * A[l++] : A[r] * A[r--];
    }
    return re;
  }
};

File: Companies/Uber/SquaresofSortedArray.java
package Companies.Uber;

public class SquaresofSortedArray {
    public int[] sortedSquares(int[] A) {
        int[] re = new int[A.length];
        int l = 0, r = A.length-1;
        for (int i = A.length-1; i >= 0; i--) {
            if (Math.abs(A[l]) > Math.abs(A[r])) {
                re[i] = A[l]*A[l];
                l++;
            } else {
                re[i] = A[r]*A[r];
                r--;
            }
        }
        return re;
    }
}
